File: Algorithms/791. Custom Sort String.rs
use std::collections::HashMap;

impl Solution {
    pub fn custom_sort_string(order: String, str: String) -> String {
        let mut hm: HashMap<char, usize> = HashMap::new();
        for (i, c) in order.char_indices() {
            *hm.entry(c).or_default() = i;
        }
        let mut t: Vec<char> = str.chars().collect();
        t.sort_unstable_by_key(|c| hm.get(c).unwrap_or(&26));
        t.iter().collect()
    }
}

File: Algorithms/791. Custom Sort String.ts
function customSortString(order: string, s: string): string {
  const chars = order.split("");
  const result = s.split("");
  result.sort(
    (a, b) => chars.findIndex((e) => e === b) - chars.findIndex((e) => e === a)
  );
  return result.reverse().join("");
}
